File: frontend/src/types/notification.ts
import { NotificationType } from './enums';

export interface NotificationDto {
  id: string;
  userId?: string;
  type: NotificationType;
  title: string;
  message: string;
  resourceId?: string;
  resourceType?: string;
  actionUrl?: string;
  isRead?: boolean;
  readAt?: string;
  isDeleted?: boolean;
  createdAt: string;
}

export interface UnreadCountDto {
  count?: number;
  unreadCount?: number;
}

export interface NotificationListResponse {
  items: NotificationDto[];
  total: number;
  unreadCount: number;
  page: number;
  pageSize: number;
}

File: src/Domain/Entities/Notification.cs
using StartupConnect.Domain.Enums;

namespace StartupConnect.Domain.Entities;

public sealed class Notification : BaseEntity
{
    public Guid UserId { get; set; }

    public User User { get; set; } = null!;

    public NotificationType Type { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Message { get; set; } = string.Empty;

    public Guid? ResourceId { get; set; }

    public string? ResourceType { get; set; }

    public string? ActionUrl { get; set; }

    public DateTimeOffset? ReadAt { get; set; }

    public bool IsDeleted { get; set; }
}
